packages/xod-arduino/platform/runtime.cpp

```diff
@@ -346,6 +347,19 @@ void runTransaction() {
     XOD_TRACE_F("Transaction started, t=");
     XOD_TRACE_LN(g_transactionTime);
 
+    // defer-* nodes are always at the very bottom of the graph,
+    // so no one will recieve values emitted by them.
+    // We must evaluate them before everybody else
+    // to give them a chance to emit values.
+    for (NodeId nid : g_defer_node_ids) {
+        if (isNodeDirty(nid)) {
+            evaluateNode(nid);
+            // clear node dirty flag, so it will evaluate
+            // on "regular" pass only if it has a dirty input
+            g_dirtyFlags[nid] &= 0x11111110;
```

Comment:
If I understand correctly, you can safely do:

```cpp
g_dirtyFlags[nid] = 0;
```

instead.

BTW, if we’d guarantee all `defer-*` nodes are at the tail of the list, you could use a single `memset` instead of the `for` loop.